### PR TITLE
Advanced Light Replacer

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Added a new janitorial item that can be researched. The advanced light replacer.
CL: [New] Lights can now have their colors and intensity changed using an advanced light replacer.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
review pls

File: UnityProject/Assets/Scripts/Items/Tool/AdvancedLightReplacer.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+		public string HoverTip()
+		{
+			StringBuilder statusText = new StringBuilder();
+			statusText.AppendLine($"Lighter Tuner On: {lightTuner}");
+			statusText.AppendLine($"Slots used: {storage.GetOccupiedSlots().Count} out of {storage.ItemStorageStructure.IndexedSlots}");
+			return statusText.ToString();
+		}
+
+		public string CustomTitle() { return null; }
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Tool/AdvancedLightReplacer.cs]: Use arrow syntax if you are going to return in the same line

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Tool/AdvancedLightReplacer.cs]: done

File: UnityProject/Assets/Standard Assets/HSVPicker/UI/ColorPicker.cs [hunk 2]
```diff
@@ -195,6 +204,12 @@ private float A
         }
     }
 
+    public void EnablePicker(Action<Color> onChange)
+    {
+	    DynamicValueChangeEvent += onChange;
```
Comment (Bod9001) [inline UnityProject/Assets/Standard Assets/HSVPicker/UI/ColorPicker.cs:209]: Make sure to unsubscribe from this if the object is destroyed or when you finish interacting

**Comment by corp-0** [comment]
you have 1 codacy issue to solve. Rename the property so it has a proper name, isn't it the ``MountState``?

File: UnityProject/Assets/Standard Assets/HSVPicker/UI/ColorPicker.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Standard Assets/HSVPicker/UI/ColorPicker.cs:209]: I'm already doing that

**Comment by MaxIsJoe** [comment]
done

**Review by corp-0 — APPROVED** [review]

**Pull request merged**